### PR TITLE
Warpx field name improvements

### PR DESCRIPTION
**Pull request by atmyers**
This makes a number of bug fixes / improvements to the field name conventions in the WarpX frontend:

1. Additional aliases have been set up, for consistency with the OpenPMD conventions.
2. The default "fluid type" for WarpX has been changed to `'mesh'` instead of `'gas'`, and some minor generalizations to the `FieldInfoContainer` have been made to allow this.
3. Most significantly, a bug for the particle velocity fields has been fixed. The thing we were interpreting as the velocity is really the relativistic momentum. This change will probably break the tests, but the new behavior is what we want.

### CODE REVIEW

**Comment by ngoldbaum** [comment]
@atmyers it looks like there are a number of test failures due changes in answers. To make this mergeable you're going to need to update those answers. I think the failure on travis was spurious and is ignorable.

**Comment by atmyers** [comment]
I'm not able to reproduce this last test failure locally, for whatever reason.

**Comment by ngoldbaum** [comment]
I'm not able to reproduce locally either. I'm going to trigger another test run and if the failure is reproducible I guess it has something to do with the order that the tests are happening in? Things to try are disabling dataset caching (grep the codebase for `_skip_cache` for more details) and logging into the test node to manually run the boxlib answer tests there.

**Comment by ngoldbaum** [comment]
@atmyers did you get a chance to reproduce this locally using the invocation I found on the test machine? I'm happy to dive in a little bit deeper to debug on the test machine too.

**Comment by atmyers** [comment]
I haven't had a chance, no. I wasn't able to read in the answers you provided on my local machine, and when I tried to generate them locally, the failure didn't happen.

**Comment by Xarthisius** [comment]
@atmyers could you add the following change and see if it helps? 

```
diff --git a/tests/nose_runner.py b/tests/nose_runner.py
index 8bf12fd96..9452e1417 100644
--- a/tests/nose_runner.py
+++ b/tests/nose_runner.py
@@ -75,7 +75,8 @@ def generate_tasks_input():
     tests = yaml.load(data)
 
     base_argv = ['--local-dir=%s' % answers_dir, '-s', '--nologcapture',
-                 '--with-answer-testing', '--answer-big-data', '--local']
+                 '--with-answer-testing', '--answer-big-data', '--local',
+                 '--with-isolation']
     args = []
 
     for test in list(tests["other_tests"].keys()):
```

**Comment by Xarthisius** [comment]
Meh, sorry it didn't help. Additional failures are interesting though. I wonder if we have some sort of implicit deps.

**Comment by ngoldbaum** [comment]
That is super weird. It has something to do with astropy's logger? We should open a separate issue about that to debug later. For now Andrew why don't you back out that last commit and let's try to figure out why the failure is happening on the test box. I expect it's a corner case and is a real bug, although something that only happens if you load a bunch of boxlib datasets in succession.

I'm going to try to reproduce this on the test box and then debug there and will report back if I find anything. Sorry this is so hard to merge! I know this improves correctness for AMReX data and it's high on my priority list to get it merged.

File: yt/frontends/boxlib/fields.py [hunk 1]
```diff
@@ -55,19 +55,19 @@ def _temperature(field, data):
...
-        ("Ez", ("V/m", ["electric_field_z"], None)),
-        ("jx", ("A", ["current_x"], None)),
-        ("jy", ("A", ["current_y"], None)),
-        ("jz", ("A", ["current_z"], None)),
+        ("Bx", ("T",   ["magnetic_field_x", "B_x"], None)),
+        ("By", ("T",   ["magnetic_field_y", "B_y"], None)),
+        ("Bz", ("T",   ["magnetic_field_z", "B_z"], None)),
+        ("Ex", ("V/m", ["electric_field_x", "E_x"], None)),
+        ("Ey", ("V/m", ["electric_field_y", "E_y"], None)),
+        ("Ez", ("V/m", ["electric_field_z". "E_z"], None)),
```
Comment (ngoldbaum) [inline yt/frontends/boxlib/fields.py]: need a comma here instead of a period

Comment (ngoldbaum) [reply yt/frontends/boxlib/fields.py]: After fixing that (no idea offhand why we weren't getting syntax errors?) I'm now getting "no old answer available" errors. So I think you just need to recreate the answers after fixing that typo and you should be good to go.

**Comment by ngoldbaum** [comment]
The test failures are due to dataset caching and are actually due to a bug in how some tests handle the dataset caching config option, see #1643.

Here’s a script that illustrates the issue:

```
import yt
from yt.config import ytcfg
ytcfg["yt", "skip_dataset_cache"] = "False"
ds = yt.load("Laser/plt00015")
ds.index
print(ds.fluid_types)
ds = yt.load("Laser/plt00015")
ds.index
print(ds.fluid_types)
```

The second print statement won’t include `”raw”`.
